google-cloud-storage/src/test/java/com/google/cloud/storage/DefaultBufferedWritableByteChannelTest.java

```diff
@@ -0,0 +1,549 @@
+   *                 7     14     21     28     35     42     49     56     63     70     77     84     91     98    105
+   * writes:  |------|------|------|------|------|------|------|------|------|------|------|------|------|------|------|
+   *                        15             30             45             60             75             90            105
+   * flushes: |--------------|--------------|--------------|--------------|--------------|--------------|--------------|
+   * </pre>
+   */
+  @Example
+  void writeOpsOfGeneratesAccurately_1() {
+    byte[] bytes = DataGenerator.base64Characters().genBytes(105);
+    ImmutableList<ByteBuffer> writes =
+        ImmutableList.of(
+            ByteBuffer.wrap(bytes, 0, 7),
+            ByteBuffer.wrap(bytes, 7, 7),
+            ByteBuffer.wrap(bytes, 14, 7),
+            ByteBuffer.wrap(bytes, 21, 7),
+            ByteBuffer.wrap(bytes, 28, 7),
+            ByteBuffer.wrap(bytes, 35, 7),
+            ByteBuffer.wrap(bytes, 42, 7),
+            ByteBuffer.wrap(bytes, 49, 7),
+            ByteBuffer.wrap(bytes, 56, 7),
+            ByteBuffer.wrap(bytes, 63, 7),
+            ByteBuffer.wrap(bytes, 70, 7),
+            ByteBuffer.wrap(bytes, 77, 7),
+            ByteBuffer.wrap(bytes, 84, 7),
+            ByteBuffer.wrap(bytes, 91, 7),
+            ByteBuffer.wrap(bytes, 98, 7));
+    ImmutableList<Long> flushes = ImmutableList.of(15L, 30L, 45L, 60L, 75L, 90L, 105L);
+    String z = "[0x00000007 * 0x0000000f]";
+    WriteOps expected = new WriteOps(bytes, 15, 7, writes, flushes, z);
+    assertThat(WriteOps.of(105, 15, 7)).isEqualTo(expected);
+  }
+
+  /**
+   *
+   *
+   * <pre>
+   *           0                                                          61
+   * data:    |------------------------------------------------------------|
+   *                         16         (16) 32         (16) 48      (13) 61
+   * writes:  |---------------|---------------|---------------|------------|
+   *             3  6  9 12 15 18 21 24 27 30 33 36 39 42 45 48 51 54 57 60
+   * flushes: |--|--|--|--|--|--|--|--|--|--|--|--|--|--|--|--|--|--|--|--||
+   * </pre>
+   */
+  @Example
+  void writeOpsOfGeneratesAccurately_2() {
+    byte[] bytes = DataGenerator.base64Characters().genBytes(61);
+    ImmutableList<ByteBuffer> writes =
+        ImmutableList.of(
+            ByteBuffer.wrap(bytes, 0, 16),
+            ByteBuffer.wrap(bytes, 16, 16),
+            ByteBuffer.wrap(bytes, 32, 16),
+            ByteBuffer.wrap(bytes, 48, 13));
+    ImmutableList<Long> flushes =
+        ImmutableList.of(
+            3L, 6L, 9L, 12L, 15L, 18L, 21L, 24L, 27L, 30L, 33L, 36L, 39L, 42L, 45L, 48L, 51L, 54L,
+            57L, 60L, 61L);
+    String z = "[0x00000010 * 0x00000003, 0x0000000d]";
+    WriteOps expected = new WriteOps(bytes, 3, 16, writes, flushes, z);
+    WriteOps actual = WriteOps.of(61, 3, 16);
```

Comment:
I'm going to ask for variable for these `constants` at least; the others are clear from the data pipeline above.

It would help to know that `61` is data size, `3` is buffer size, and `16` is writeSize without having to pull up definition of WriteOps.